isic/core/api/image.py

```diff
@@ -84,17 +90,30 @@ def list_images(request: HttpRequest):
 
 @router.get(
     "/search/",
-    response=list[ImageOut],
+    response={200: list[ImageOut], 400: dict},
     summary="Search images with a key:value query string.",
     description=render_to_string("core/swagger_image_search_description.html"),
 )
 @paginate(CursorPagination)
 def search_images(request: HttpRequest, search: SearchQueryIn = Query(...)):  # noqa: B008
-    return search.to_queryset(user=request.user, qs=default_qs)
+    try:
+        return search.to_queryset(user=request.user, qs=default_qs)
+    except ParseException:
+        # Normally we'd like this to be handled by the input serializer validation, but
+        # for backwards compatibility we must return 400 rather than 422.
+        # The pagination wrapper means we can't just return the response we'd like from here.
+        # The handler for this exception type is defined in urls.py.
+        raise ImageSearchParseError()
 
 
 @router.get("/facets/", response=dict, include_in_schema=False)
 def get_facets(request: HttpRequest, search: SearchQueryIn = Query(...)):  # noqa: B008
+    if search.query:
+        try:
+            parse_query(search.query)
+        except ParseException:
+            raise ImageSearchParseError()
```

Comment:
This doesn't use pagination, so we could just return a response, but I figured it'd be more consistent to use the same error handler as the other endpoint since it exists.